cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/DatabaseBrowser/index.js

```diff
@@ -97,48 +113,47 @@ export default class DatabaseBrowser extends Component {
         }
       );
   }
+
   fetchTables() {
-    let {userName, password, connectionString} = this.state.properties;
-    if (isEmpty(userName) || isEmpty(password) || isEmpty(connectionString)) {
-      return;
-    }
-    let {selectedNamespace: namespace} = NamespaceStore.getState();
+    if (!this.state.connectionId) { return null; }
+
+    let namespace = NamespaceStore.getState().selectedNamespace;
     let params = {
-      namespace
+      namespace,
+      connectionId: this.state.connectionId
     };
-    DataPrepApi
-      .listTables(params, {userName, password, connectionString})
+
+    DataPrepApi.listTables(params)
       .subscribe(
         (res) => {
           this.setState({
-            tables: res,
+            tables: res.values,
             loading: false
           });
         },
         (err) => {
           this.setState({
-            error: typeof err === 'object' ? err.response : err,
+            error: typeof err === 'object' ? err.response.message : err,
             loading: false
           });
         }
       );
   }
+
   render() {
     // FIXME: This should be going away as user wouldn't even be reaching here if plugin is not installed
```

Comment:
nit: We could remove this fixme as we know a usecase where the user opens a connection while the plugin is uninstalled somehow. This would more likely happen if the driver was deleted after the connection was created.